sentry-spring-boot-starter/src/main/java/io/sentry/spring/autoconfigure/SentryProperties.java

```diff
@@ -0,0 +1,158 @@
+package io.sentry.spring.autoconfigure;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Configuration for Sentry, example:
+ *
+ * <pre>
+ * sentry:
+ *     enabled: true
+ *     init-default-client: true
+ *     dsn: https://00059966e6224d03a77ea5eca10fbe18@sentry.mycompany.com/14
+ *     release: "1.0.1"
+ *     dist: x86
+ *     environment: staging
+ *     serverName: megaServer
+ *     tags:
+ *         firstTag: Hello
+ *         secondTag: Awesome
+ *     mdcTags: [mdcTagA, mdcTagB]
+ *     extra:
+ *         extraTag: extra
+ *     options:
+ *         stacktrace.app.packages: com.mycompany,com.other.name
+ *         sample.rate: 0.75
+ *         uncaught.handler.enabled: false
+ * </pre>
+ */
+@ConfigurationProperties("sentry")
+public class SentryProperties {
+
+    /**
+     * Data source name
+     * All of the options can be configured by setting querystring parameters on the DSN itself.
+     * https://docs.sentry.io/clients/java/config/#configuration-via-the-dsn
+     * More information about configuration via DSN https://docs.sentry.io/clients/java/config/#configuration-via-the-dsn
+     */
+    private String dsn;
+
+    /**
+     * The application version that will be sent with each event.
+     */
+    private String release;
+
+    /**
+     * The application distribution that will be sent with each event.
+     * Note that the distribution is only useful (and used) if the release is also set.
+     */
+    private String dist;
+
+    /**
+     * The application environment that will be sent with each event.
+     */
+    private String environment;
+
+    /**
+     * The server name that will be sent with each event.
+     */
+    private String serverName;
+
+    /**
+     * Tags that will be sent with each event.
+     */
+    private Map<String, String> tags = new LinkedHashMap<>();
+
+    /**
+     * Set tag names that are extracted from the SLF4J MDC system.
+     */
+    private Set<String> mdcTags = new HashSet<>();
+
+    /**
+     * Set extra data that will be sent with each event (but not as tags).
+     */
+    private Map<String, Object> extra = new LinkedHashMap<>();
+
+    /**
+     * Additional options, check the <a href="https://docs.sentry.io/clients/java/config/">documentation</a>.
+     */
+    private Map<String, String> options = new LinkedHashMap<>();
```

Comment:
Hmm... Do you think this is good idea to separate some properties to `SentryProperties` and some properties to `options` block? I think, it would be better, if you add all properties as `SentryProperties` attributes. With attributes you can set correct property types.

Reply:
Can you help me with this? You can make a PR, I'll merge it to my master: https://github.com/donbeave/sentry-java/commits/master

Reply:
@donbeave What do you think about this config:
```java
public class SentryProperties {

    private String release;
    private String dist;
    private String environment;
    private String serverName;
    private Set<String> tags;
    private Set<String> mdcTags;

    private Map<String, String> extra;

    private Boolean hideCommon;
    private Boolean compression;
    private Integer maxMessageLength;
    private Duration timeout;

    private Double sampleRate;
    private Boolean uncaughtHandlerEnabled;

    private StacktraceProperties stacktrace;
    private BufferProperties buffer;
    private AsyncProperties async;

    private static class StacktraceProperties {

        private Boolean hideCommon;
        private Set<String> appPackages;
    }

    private static class BufferProperties {

        private String dir;
        private Integer size;
        private Duration flushtime;
        private Duration shutdownTimeout;
        private Boolean gracefulShutdown;
    }

    private static class AsyncProperties {

        // async.enabled maps to async Sentry property
        private Boolean enabled;
        private Duration shutdownTimeout;
        private Boolean gracefulShutdown;
        private Integer queueSize;
        private Integer threads;
        private Integer priority;
    }
}
```

Just need to add normal mapping spring properties to sentry properties.

Reply:
Looks good to me.

Reply:
@altro3 will you make a PR with this config?

Reply:
I'll do it, `Duration flushTime` and `Duration shutdownTimeout` I'll replace with `Integer`, because `sentry-java` are using Java 1.7.